### PR TITLE
IMP fix default when block for filter

### PR DESCRIPTION
**Pull request by bd16s**
When creating resource filter without passing a `when` block, it automatically adds an empty one which is not the expected behavior. This is a proposed fix to the issue.

### CODE REVIEW

File: sensu/resource_filter.go [hunk 1]
```diff
@@ -56,8 +56,13 @@ func resourceFilterCreate(d *schema.ResourceData, meta interface{}) error {
...
 	runtimeAssets := expandStringList(d.Get("runtime_assets").([]interface{}))
-	when := expandTimeWindows(d.Get("when").(*schema.Set).List())
-
+	// to prevent the creation of an empty when block
+	whenRaw, whenExists := d.GetOk("when")
+	var when *types.TimeWindowWhen = nil
+	if whenExists {
+		expandedTimeWindows := expandTimeWindows(whenRaw.(*schema.Set).List())
+		when = &expandedTimeWindows
+	}
```
Comment (mdelaney) [inline sensu/resource_filter.go:65]: nit, add blank line below

File: sensu/resource_filter.go [hunk 2]
```diff
@@ -56,8 +56,13 @@ func resourceFilterCreate(d *schema.ResourceData, meta interface{}) error {
 
 	expressions := expandStringList(d.Get("expressions").([]interface{}))
 	runtimeAssets := expandStringList(d.Get("runtime_assets").([]interface{}))
-	when := expandTimeWindows(d.Get("when").(*schema.Set).List())
-
+	// to prevent the creation of an empty when block
```
Comment (mdelaney) [inline sensu/resource_filter.go:59]: readability nit: add blank line above

**Review by mdelaney — COMMENTED** [review]
couple of super picky comments, other than that, are there any tests that should be added to assert that if `when` is not specified it doesn't end up creating a default that's used instead?

**Comment by bd16s** [comment]
This issue is fixed in upstream v0.8.0, closing.

**Pull request closed without merging**